execute/transformation.go

```diff
@@ -9,11 +9,17 @@ import (
 	"github.com/influxdata/flux/plan"
 )
 
+// Transformation represents functions that control of how the execution
+// engine will take in a flux Table, perform the necessary data processing
+// and return an output flux Table
 type Transformation interface {
 	RetractTable(id DatasetID, key flux.GroupKey) error
+	// Process takes in a flux Table, performs data processing on it
+	// and produces an output flux Table
 	Process(id DatasetID, tbl flux.Table) error
```

Comment:
This is correct in concept but the method signature doesn't really communicate the same thing. In other words, if the function produces an output table where is it? Its not on the function signature. Can you explain where the output table is created?

Reply:
When I look at the `createToTransformation` function in the `experimental` package in IDPE, it returns a `Dataset` and a `Transformation`.  Based on this, and the function signature that you pointed out, it seems that the functions in the `Transformation` interface only return nil/error. I _think_ that the working information for the `Transformation` is stored in a `Dataset` and is returned by the function that creates the `Transformation` and the `Dataset`. In this case, `createToTransformation`.

Reply:
@nathanielc I know that this isn't quite right, but I think I'm understanding the code a little bit better. Let me know what you think and I'll update it.

Reply:
@fchikwekwe That is correct, the Process method is responsible for creating a new output table and adding it to the dataset.